scripts/generate_metrics_scripts.py

```diff
@@ -27,36 +38,63 @@ def main(
     combinations = [
         f"--experiment_groups_path {experiment_groups_path} "
         + f"--experiment_group {experiment_group} "
-        + f"--dmpair_config {dmpair_config_path} "
-        + f"--metrics_config {metrics_config_path} "
+        + f"--dmpair_config_path {dmpair_config_path} "
+        + f"--metrics_config_path {metrics_config_path} "
+        + f"--metrics_file {metrics_file} "
+        + f"--trainer_config_path {trainer_config_path} "
         + f"--seed_index {seed_index} "
         + f"--dataset_index {dataset_index}"
         for seed_index in range(NUM_SEEDS)
         for dataset_index in range(NUM_PAIRS)
     ]
 
     # Prepare path
-    scripts_path = os.path.join(constants.PROJECT_ROOT, "metrics_scripts")
+    scripts_path = os.path.join(constants.PROJECT_ROOT, "metrics_scripts", metrics_file)
 
     # If bash scripts path does not exist, create it
     if not os.path.isdir(scripts_path):
         os.mkdir(scripts_path)
 
     # Generate files + script names
-    experiment_pair_names = [
-        scripts_path
-        + f"/{experiment_group}_{dataset_index}_{seed_index}"
-        + "_metrics.sh"
+    experiment_names = [
+        f"{experiment_group}_{dataset_index}_{seed_index}"
         for seed_index in range(NUM_SEEDS)
         for dataset_index in range(NUM_PAIRS)
     ]
+    script_names = [
+        os.path.join(scripts_path, experiment_name + "_metrics.sh")
+        for experiment_name in experiment_names
+    ]
+
+    # Are we using baskerville?
+    use_baskerville = metrics_config["baskerville"]
+
+    # If yes, prepare jinja template
+    if use_baskerville:
```

Comment:
we're doing this condition check twice (now and l86, inside loop). It also defines new variables that are then only used later after condition
I suggest restructure to two separate methods from this point:
`write_baskerville_scripts` and `write_default_scripts`, or similar. Each of these methods would loop through the combos and do the writes

Reply:
Have factored these out into utils, and called the Baskerville one in model training and attacking too